SimTKcommon/Simulation/src/State.cpp

```diff
@@ -312,7 +313,8 @@ StateImpl::StateImpl(const StateImpl& src)
 :   currentSystemStage(Stage::Empty)
 {
     initializeStageVersions();
-
+    stateLock = new std::mutex; //TODO: When we copy a state, do we want it to have a different stateLock? I think yes?
```

Comment:
Okay. I originally declared the mutex on the heap because I couldn't lock the mutex in a const method. The places that I lock the mutex are no longer const so I can declare it on the stack. The only concern is that if we want to lock const methods in the future - Is there another way to do that?